### PR TITLE
Refactoring Code

### PR DESCRIPTION
**Pull request by khizirsiddiqui**
This shall clean a lot of redundant code.

Remaining:
 - Tests for models LeNe, ModLeNet, NIN and Shallow

@Het-Shah and @NeelayS should I remove the `teacher` and `student` from `orignal`.

### CODE REVIEW

**Comment by Het-Shah** [comment]
> @Het-Shah and @NeelayS should I remove the teacher and student from orignal

yea seems like we should remove it to me.

File: KD_Lib/models/nin.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
+from torch import nn
+
+
+class NIN(nn.Module):
+    def __init__(self, num_classes=10, in_channels=3):
```
Comment (Het-Shah) [inline KD_Lib/models/nin.py:5]: maybe we can give a more meaningful name to this.

Comment (khizirsiddiqui) [reply KD_Lib/models/nin.py:5]: Ok, will rename it. :+1:

**Comment by khizirsiddiqui** [comment]
Maybe we should move it another independent class. It's a proper method in itself, though it primarily introduces a loss metric only.

**Comment by khizirsiddiqui** [comment]
Het, I see `self.loss` in `BaseClass` assigned but is not used except in `orignal`.

**Comment by Het-Shah** [comment]
> Het, I see self.loss in BaseClass assigned but is not used except in orignal.
Initially i was intending to keep original inside BaseClass then changed my mind. Doesn't matter i guess we can move it out of BaseClass.

File: KD_Lib/noisy/utils.py [hunk 2]
```diff
@@ -1,24 +1,5 @@
...
-            output = model(data)
-            pred = output.argmax(dim=1, keepdim=True)
-            correct += pred.eq(target.view_as(pred)).sum().item()
-
-    print(f'{correct}/{total} correct')
-    print(f'The accuracy of the model is {correct/total}')
-
-
 def add_noise(x, variance=0.1):
     return x * (1 + (variance**0.5) * torch.randn_like(x))
```
Comment (Het-Shah) [inline KD_Lib/noisy/utils.py:5]: we can move this to the place where it is being used right? we don't need a new file just for this function.

**Comment by khizirsiddiqui** [comment]
@avishreekh is documenting the `BaseClass`, better if she changes, this will avoid merge conflicts if she has already started work.

File: KD_Lib/noisy/utils.py (hunk 2 above)
Comment (khizirsiddiqui) [reply KD_Lib/noisy/utils.py:5]: :+1: han, will do.

**Comment by Het-Shah** [comment]
Yea BaseClass we can change it easily later on.

**Comment by Het-Shah** [comment]
Merging this.
